src/extractor/guidance/intersection_handler.cpp

```diff
@@ -400,108 +393,105 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
 
     const EdgeData &in_way = node_based_graph.GetEdgeData(via_edge);
 
-    std::vector<out_way> out_ways;
-
-    // create sortable vector of only entry allowed out_ways
     // get a count of number of ways from that intersection that qualify to have
     // continue instruction because they share a name with the approaching way
     std::int64_t continue_count = 0, continue_count_valid = 0;
     for (std::size_t i = 1, last = intersection.size(); i < last; ++i)
     {
-        out_way out;
-        const auto out_node = node_based_graph.GetEdgeData(intersection[i].eid);
+        const auto out_data = node_based_graph.GetEdgeData(intersection[i].eid);
 
-        out.road = &intersection[i];
-        out.same_name_id = !util::guidance::requiresNameAnnounced(
-            in_way.name_id, out_node.name_id, name_table, street_name_suffix_table);
-        if (out.same_name_id)
+        bool out_same_name_id = !util::guidance::requiresNameAnnounced(
+            in_way.name_id, out_data.name_id, name_table, street_name_suffix_table);
+        if (out_same_name_id)
             ++continue_count;
 
-        if (!out.road->entry_allowed)
-        {
-            ++continue_count_valid;
+        if (!intersection[i].entry_allowed)
             continue;
-        }
 
-        out.index = i;
-        out.deviation_from_straight = angularDeviation(intersection[i].angle, STRAIGHT_ANGLE);
-        out.same_classification = out_node.road_classification == in_way.road_classification;
-        out.same_or_higher_priority =
-            out_node.road_classification.GetPriority() <= in_way.road_classification.GetPriority();
-
-        out_ways.push_back(out);
+        ++continue_count_valid;
     }
     const std::pair<std::int64_t, std::int64_t> num_continue_names =
         std::make_pair(continue_count, continue_count_valid);
 
-    const auto End = end(out_ways);
-    // dead end, or otherwise no exitable ways
-    if (out_ways.size() == 0)
-        return 0;
-
-    // sort out ways by what they share with the in way
-    const auto sort_by = [](const out_way &lhs, const out_way &rhs) {
-        auto left_tie = std::tie(
-            rhs.deviation_from_straight, lhs.same_classification, lhs.same_or_higher_priority);
-        auto right_tie = std::tie(
-            lhs.deviation_from_straight, rhs.same_classification, rhs.same_or_higher_priority);
-        return left_tie > right_tie;
+    const auto SameName = [&](const EdgeData &way_data) {
+        return way_data.name_id == in_way.name_id;
+    };
+    auto sameOrHigherPriority = [&](const auto &way_data) {
```

Comment:
capture defaults are sometimes nice, but if it's only a single value you need, I'd prefer capturing explicitly (`[&in_way]`).

Also `way_data` vs `in_way`, both should be ways or data, right?